src/olympia/shelves/tests/test_forms.py

```diff
@@ -81,44 +73,40 @@ def test_clean_search_returns_404(self):
             'endpoint': 'search',
             'criteria': self.criteria_404}
         form = ShelfForm(data)
-        assert not form.is_valid()
-        with self.assertRaises(ValidationError) as exc:
-            form.clean()
-        assert exc.exception.message == (
-            'Check criteria - No data found')
+        form.is_valid()
```

Comment:
so, now the form _is_ valid?  Are there tests that prove that these changes still result in an error here prevent the Shelf from being saved?

Reply:
I removed the unnecessary changes to the form/tests, and added a test to confirm that `def clean` checks if there are missing required fields.